larray/io/array.py

```diff
@@ -77,22 +77,17 @@ def df_aslarray(df, sort_rows=False, sort_columns=False, raw=False, parse_header
             # take the first column which contains '\'
             # pos_last = next(i for i, v in enumerate(columns) if '\\' in str(v))
             pos_last = next(i for i, v in enumerate(columns) if isinstance(v, basestring) and '\\' in v)
-            onedim = False
         except StopIteration:
             # we assume first column will not contain data
             pos_last = 0
-            onedim = True
 
         axes_names = columns[:pos_last + 1]
-        if onedim:
-            df = df.iloc[:, 1:]
-        else:
-            # This is required to handle int column names (otherwise we can simply use column positions in set_index).
-            # This is NOT the same as df.columns[list(range(...))] !
-            index_columns = [df.columns[i] for i in range(pos_last + 1)]
-            # TODO: we should pass a flag to df_aslarray so that we can use inplace=True here
-            # df.set_index(index_columns, inplace=True)
-            df = df.set_index(index_columns)
+        # This is required to handle int column names (otherwise we can simply use column positions in set_index).
+        # This is NOT the same as df.columns[list(range(...))] !
+        index_columns = [df.columns[i] for i in range(pos_last + 1)]
+        # TODO: we should pass a flag to df_aslarray so that we can use inplace=True here
+        # df.set_index(index_columns, inplace=True)
+        df = df.set_index(index_columns)
```

Comment:
I am surprised this works for 1D arrays. I wonder if the result wouldn't be subtly different, so I am nervous to introduce this in a bugfix release. At a minimum, we will need to check how user models behave with this change, so maybe it would be wise to postpone this to 0.26?